tests/tests.rs

```diff
@@ -44,25 +158,259 @@ fn decode_rfc4648_4() {
 }
 
 #[test]
+fn decode_rfc4648_4_no_padding() {
+    compare_decode("foob", "Zm9vYg");
+}
+
+#[test]
 fn decode_rfc4648_5() {
     compare_decode("fooba", "Zm9vYmE=");
 }
 
 #[test]
+fn decode_rfc4648_5_no_padding() {
+    compare_decode("fooba", "Zm9vYmE");
+}
+
+#[test]
 fn decode_rfc4648_6() {
     compare_decode("foobar", "Zm9vYmFy");
 }
 
-//this is a MAY in the rfc
+//this is a MAY in the rfc: https://tools.ietf.org/html/rfc4648#section-3.3
 #[test]
-fn decode_allow_extra_pad() {
-    compare_decode("alice", "YWxpY2U=====");
+fn decode_pad_inside_fast_loop_chunk_error() {
+    // can't PartialEq Base64Error, so we do this the hard way
+    match decode("YWxpY2U=====").unwrap_err() {
+        Base64Error::InvalidByte(offset, byte) => {
+            // since the first 8 bytes are handled in the fast loop, the
+            // padding is an error. Could argue that the *next* padding
+            // byte is technically the first erroneous one, but reporting
+            // that accurately is more complex and probably nobody cares
+            assert_eq!(7, offset);
```

Comment:
Slightly janky error detection when there's so much extra padding that it spills into the fast loop